### PR TITLE
Api create model method

### LINKED ISSUE #568: Deprecate `create_model`
I propose to turn the `hspy.create_model` function into a `Signal` method that should allow us to write:

```python
 m = s.create_mode()

```

Instead of 

```python

m = create_model(s)

```

The advantage is that, given that there are options which are specific for each `Signal` type, the docstring and argument list will be less confusing. What do you think?

### PR DESCRIPTION
**Pull request by francisco-dlp**
Fixes #568

### CODE REVIEW

File: hyperspy/tests/model/test_eelsmodel.py [hunk 1]
```diff
@@ -4,6 +4,53 @@
...
+    def test_auto_add_edges_true(self):
+        m = self.s.create_model(auto_add_edges=True)
+        cnames = [component.name for component in m]
+        nose.tools.assert_true("B_K" in cnames and "C_K" in cnames)
+
+    def test_gos(self):
+        m = self.s.create_model(auto_add_edges=True, GOS="hydrogenic")
+        nose.tools.assert_true(m["B_K"].GOS._name == "hydrogenic")
+
+    def test_auto_add_edges_true(self):
```
Comment (vidartf) [inline hyperspy/tests/model/test_eelsmodel.py]: Should be renamed to `test_auto_add_background_true`.

Comment (francisco-dlp) [inline hyperspy/tests/model/test_eelsmodel.py]: Fixed, thanks!

File: doc/user_guide/eels.rst [hunk 2]
```diff
@@ -119,29 +119,29 @@ Set some important experimental information that is missing from the original
...
 
 .. code-block:: python
-       
+
     >>> s.add_elements(('B', 'N'))
-    
-    
+
+
 We pass the low-loss spectrum to :py:func:`~.hspy.create_model` to include the
```
Comment (vidartf) [inline doc/user_guide/eels.rst]: This one should be updated. (:py:func:`~.hspy.create_model`)

File: hyperspy/hspy.py [hunk 3]
```diff
@@ -97,11 +97,9 @@ def create_model(signal, *args, **kwargs):
     A Model class
```
Comment (vidartf) [inline hyperspy/hspy.py]: Docstring of hspy.py still has docs for `create_model`, should be removed.

File: hyperspy/tests/model/test_chi_squared.py [hunk 4]
```diff
@@ -28,7 +28,7 @@ class TestChiSquared:
```
Comment (vidartf) [inline hyperspy/tests/model/test_chi_squared.py]: Import `from hyperspy.hspy import create_model` unused in this file.

File: hyperspy/tests/model/test_fit_component.py [hunk 5]
```diff
@@ -34,7 +34,7 @@ def setUp(self):
         g.sigma.value = 500.0
```
Comment (vidartf) [inline hyperspy/tests/model/test_fit_component.py]: Import `from hyperspy.hspy import create_model` unused in this file.

File: hyperspy/tests/model/test_set_parameter_state.py [hunk 6]
```diff
@@ -31,7 +31,7 @@ def setUp(self):
         g2 = Gaussian()
```
Comment (vidartf) [inline hyperspy/tests/model/test_set_parameter_state.py]: Import `from hyperspy.hspy import create_model` unused in this file.

File: hyperspy/tests/model/test_set_parameter_value.py [hunk 7]
```diff
@@ -31,7 +31,7 @@ def setUp(self):
         g2 = Gaussian()
```
Comment (vidartf) [inline hyperspy/tests/model/test_set_parameter_value.py]: Import `from hyperspy.hspy import create_model` unused in this file.

File: hyperspy/hspy.py [hunk 8]
```diff
@@ -97,11 +97,9 @@ def create_model(signal, *args, **kwargs):
...
-    from hyperspy.model import Model
-    if isinstance(signal, EELSSpectrum):
-        return EELSModel(signal, *args, **kwargs)
-    else:
-        return Model(signal, *args, **kwargs)
+    import warnings
+    warnings.warn(
+        "This function is deprecated and will be removed in HyperSpy 0.9. "
+        "Please use the equivalent `Signal.create_model` method "
+        "instead.", DeprecationWarning)
```
Comment (vidartf) [inline hyperspy/hspy.py]: I principle I strictly agree with using `DeprecationWarning`, however, it is currently ignored by default (i.e. not printed to console). Are we OK with that?

Comment (francisco-dlp) [inline hyperspy/hspy.py]: This issue (warnings not being printed) has been fixed in #620 (9a4c307a4b103803c4099c92c47882cbe4cc555e). I thought that the issue was small enough not to be noticed but I was wrong: I should have put that into a separate branch...

**Comment by francisco-dlp** [comment]
@vidartf, thanks for the thorough review: I've just uploaded fixes for all the issues that you have raised.

File: hyperspy/hspy.py (hunk 8 above)
Comment (vidartf) [inline hyperspy/hspy.py]: OK, so `warnings.simplefilter('default')` fixes this, but shouldn't it be put somewhere more central than those files?

Comment (francisco-dlp) [inline hyperspy/hspy.py]: I'm not sure if I get your point. I think that the issue was caused by the fact that [here](https://github.com/hyperspy/hyperspy/commit/9a4c307a4b103803c4099c92c47882cbe4cc555e#diff-e27c75df2dece36803d7c8ccc9166673L11) the filter default was changed to "ignore" and never changed back to "default".

Comment (vidartf) [inline hyperspy/hspy.py]: Case in point: When I was testing out this PR, I never got a warning while using `hspy.create_model`

Comment (vidartf) [inline hyperspy/hspy.py]: I think I see the problem: DeprecationWarnings are normally ignored, however when setting the filter to `default`: "This enables default handling for all warnings, including those that are ignored by default". This is quite counter intuitive, but...

Comment (vidartf) [inline hyperspy/hspy.py]: Ref: https://docs.python.org/2/library/warnings.html#updating-code-for-new-versions-of-python

Comment (vidartf) [inline hyperspy/hspy.py]: Proposed solution: Run `warnings.simplefilter('default')` somewhere (e.g. hyperspy.hspy).

Comment (vidartf) [inline hyperspy/hspy.py]: I think I'll make this a separate issue, and merge this. Agreed?

Comment (francisco-dlp) [inline hyperspy/hspy.py]: Yes, I think that the issue is more general, so I agree that it'll be
better to let this in and fix the general issue in a different PR.
An alternative would be to use numpy.VisibleDeprecationWarning.

**Comment by vidartf** [comment]
PS: Strange that the landscape bot didn't pick up on the unused imports. Is that not something it checks for?

**Comment by francisco-dlp** [comment]
Yes, that's strange: maybe the `create_model` method calls confused it. Anyway, with you who needs landscape ;) ?

**Comment by vidartf** [comment]
> with you who needs landscape ;) ?

I'd recommend everyone to use an IDE with integrated `PEP8`, `pyflakes`, and possibly `pylint`.

**Pull request merged**